### ISSUE TITLE
Type inheritance causes missing imports

### ISSUE THREAD

**Issue by ian-speers** [atomicbits/scraml#146]
It seems like there is an issue with inheritance wherein child classes may be generated without all necessary imports. e.g.
```raml
#%RAML 1.0
title: Test

types:
  Person:
    type: object
    properties:
      hobbies: string[]
  Adult:
    type: Person
    properties:
      job: string
```
will generate the `Person` parent class:
```java
package scraml.com.example;

import com.fasterxml.jackson.annotation.*;

import java.util.List;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes({
  @JsonSubTypes.Type(value = Adult.class, name = "Adult"),
  @JsonSubTypes.Type(value = Person.class, name = "Person")
})
public class Person {

  @JsonProperty(value = "hobbies")
  private List<String> hobbies;

  public Person() {}

  public Person(List<String> hobbies) {
    this.setHobbies(hobbies);
  }

  public List<String> getHobbies() {
    return hobbies;
  }

  public void setHobbies(List<String> hobbies) {
    this.hobbies = hobbies;
  }
}
```
and the `Adult` child class:
```java
package scraml.com.example;

import com.fasterxml.jackson.annotation.*;

public class Adult extends Person {

  @JsonProperty(value = "job")
  private String job;

  public Adult() {}

  public Adult(List<String> hobbies, String job) {
    this.setHobbies(hobbies);
    this.setJob(job);
  }

  public String getJob() {
    return job;
  }

  public void setJob(String job) {
    this.job = job;
  }
}
```
The `Adult` class constructor requires a `List<String>` of hobbies (as specified in the parent definition). However, the `import java.util.List` statement is missing from the class file.

I set up a simple project demonstrating the issue [here](https://github.com/ian-speers/scraml-bug)

**Comment by ian-speers** [comment]
In short -- it appears that generated classes only contain imports for types mentioned in the corresponding raml type definition. They do not contain imports for types of inherited members.

**Comment by rigolepe** [comment]
I'll fix this one really soon, it's clearly a blocking bug.

**Comment by rigolepe** [comment]
Fixed in 0.7.2. All parent fields are now imported (recursively upwards to the top of the hierarchy).

**Comment by ian-speers** [comment]
Fantastic, thanks a bunch @rigolepe!!
